msticpy/context/tiproviders/mblookup.py

```diff
@@ -0,0 +1,317 @@
+                res = requests.post(_BASE_URL, data=data)
+                res = json.loads(res.text)
+                if res["query_status"] == 'ok':
+                    observable_df = json_normalize(res['data'])
+                    return observable_df
+                return res["query_status"]
+            except requests.exceptions.RequestException as err:
+                return err
+
+        elif mb_type == "gimphash":
+            try:
+                data = {'query': 'get_gimphash', "gimphash": observable, "limit": limit}
+                res = requests.post(_BASE_URL, data=data)
+                res = json.loads(res.text)
+                if res["query_status"] == 'ok':
+                    observable_df = json_normalize(res['data'])
+                    return observable_df
+                return res["query_status"]
+            except requests.exceptions.RequestException as err:
+                return err
+
+        elif mb_type == "issuerinfo":
+            try:
+                data = {'query': 'get_issuerinfo', "issuer_cn": observable}
+                res = requests.post(_BASE_URL, data=data)
+                res = json.loads(res.text)
+                if res["query_status"] == 'ok':
+                    observable_df = json_normalize(res['data'])
+                    return observable_df
+                return res["query_status"]
+            except requests.exceptions.RequestException as err:
+                return err
+
+        elif mb_type == "subjectinfo":
+            try:
+                data = {'query': 'get_subjectinfo', "subject_cn": observable}
+                res = requests.post(_BASE_URL, data=data)
+                res = json.loads(res.text)
+                if res["query_status"] == 'ok':
+                    observable_df = json_normalize(res['data'])
+                    return observable_df
+                return res["query_status"]
+            except requests.exceptions.RequestException as err:
+                return err
+
+        elif mb_type == "certificate":
+            try:
+                data = {'query': 'get_certificate', "serial_number": observable}
+                res = requests.post(_BASE_URL, data=data)
+                res = json.loads(res.text)
+                if res["query_status"] == 'ok':
+                    observable_df = json_normalize(res['data'])
+                    return observable_df
+                return res["query_status"]
+            except requests.exceptions.RequestException as err:
+                return err
+
+    def download_sample(self, sha2):
+        """ Download specified sample from MB"""
+        try:
+            data = {'query': 'get_file', "sha256_hash": sha2}
+            res = requests.post(_BASE_URL, data=data)
+            return res.content
+        except requests.exceptions.RequestException as err:
+            return err
+
+    def get_recent(self, selector):
+        """ Get the recent MB additions.
+
+            selector:

```

Comment:
Need some type hints here, plus can you please use numpy docstring format - [numpy docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html)